osu.Game/Overlays/Chat/DrawableChannel.cs

```diff
@@ -249,31 +249,32 @@ private class ChannelScrollContainer : UserTrackingScrollContainer
             /// </summary>
             private const float auto_scroll_leniency = 10f;
 
+            private bool trackNewContent = true;
             private float? lastExtent;
 
-            protected override void OnUserScroll(float value, bool animated = true, double? distanceDecay = default)
+            protected override void OnScrollChange(bool byUser)
             {
-                base.OnUserScroll(value, animated, distanceDecay);
-                lastExtent = null;
+                base.OnScrollChange(byUser);
+
+                if (byUser)
+                    lastExtent = null;
```

Comment:
I'm not sure if this should only be done on user scrolls, but I don't remember the existence of it here either... Nothing looks to regres with removing these lines specifically.

Reply:
Can you explain what this comment means? What is "the existence of it here"? Which "lines" are being "removed"? I am very confused, personally.

Reply:
The existence of `lastExtent = null` and whether to remove that because nothing regresses and tests pass.

Reply:
I wouldn't be surprised if this was related to [my other comment](https://github.com/ppy/osu/pull/17115/files#r820151287) - i.e. all of this is overengineered and all you need at the end of the day is just the `IsScrolledToEnd()` check written properly.

Reply:
I have removed this anyways and simplified to just using `!IsScrolledToEnd` instead of `extent > lastExtent`.